src/traits.rs

```diff
@@ -4,51 +4,94 @@ use core::fmt::Debug;
+    /// result of the parsing operation, with the leading `--` stripped.
+    ///
+    /// A long option is defined as an argument that follows the pattern
+    /// `--flag` or `--flag=VALUE`, where `VALUE` may be empty. For
+    /// example, `"--flag"` would parse as `Some(("flag"), None)` and
+    /// `"--flag=value"` would parse as `Some(("flag", Some("value")))`.
+    /// `"--flag="` would parse as `Some(("flag", Some("")))`.
+    fn parse_long_opt(self) -> Option<(Self, Option<Self>)>;
+
+    /// Attempts to parse this argument as a "short option cluster".
+    /// Returns the short option cluster if present.
     ///
-    /// The given "short option cluster" must be valid for at least one
-    /// [`consume_short_opt`].
-    fn short_cluster(self) -> Option<Self>;
+    /// A "short option cluster" is defined as any [`Self`] such that
+    /// either at least one [`ShortFlag`][Self::ShortFlag] can be
+    /// extracted from it using
+    /// [`consume_short_opt`][Self::consume_short_opt], or it can be
+    /// converted to a value for a preceding short option using
+    /// [`consume_short_val`][Self::consume_short_val].
+    ///
+    /// A short option cluster is signaled by the presence of a leading
+    /// `-` in an argument, and does not include the leading `-`. The
+    /// returned "short option cluster" must be valid for at least one
+    /// [`consume_short_opt`][Self::consume_short_opt] or
+    /// [`consume_short_val`][Self::consume_short_val].
+    ///
+    /// This method does not need to guard against `--` long options.
+    /// [`parse_long_opt`][Self::parse_long_opt] will be called first by
+    /// [`Options::next`][crate::Options::next].
+    fn parse_short_cluster(self) -> Option<Self>;
 
     /// Attempts to consume one short option from a "short option
-    /// cluster". Returns the short option that was consumed (if any)
-    /// and the rest of the cluster (if non-empty).
+    /// cluster", as defined by
+    /// [`parse_short_cluster`][Self::parse_short_cluster]. Returns the
+    /// short option that was consumed and the rest of the cluster (if
+    /// non-empty).
+    ///
+    /// The returned cluster is subject to the same requirements as the
+    /// return value of
+    /// [`parse_short_cluster`][Self::parse_short_cluster]; namely, its
+    /// validity for [`consume_short_opt`][Self::consume_short_opt] or
+    /// [`consume_short_val`][Self::consume_short_val].
     fn consume_short_opt(self) -> (Self::ShortOpt, Option<Self>);
 
-    /// Attempts to consume the value of a short option from a "short
-    /// option cluster". Returns the value that was consumed, if any.
-    fn consume_short_val(self) -> Option<Self>;
+    /// Consumes the value of a short option from a "short
+    /// option cluster", as defined by
+    /// [`parse_short_cluster`][Self::parse_short_cluster]. Returns the
+    /// value that was consumed.
+    fn consume_short_val(self) -> Self;
```

Comment:
Semantically, yes. Technically, `getargs` could avoid calling this method since all implementations of `Argument` *by `getargs`* unconditionally return `self`, but I think that would limit downstream flexibility for things that are not simple strings (could be objects, for example).